.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,51 @@
+name: PR Workflow
+
+on:
+  pull_request:
+    branches: ['**']
+
+jobs:
+  Code-Quality:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+          cache: 'npm'
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Lint
+        run: npm run lint
+
+      - name: Type Check
+        run: npm run typecheck
+
+      - name: Run Tests with Coverage
+        run: npm run test:coverage
+        
+
+      # will uncomment once the normal workflow start working smoothly
+      # - name: Merge Coverage Reports
+      #   run: npx lcov-result-merger 'apps/*/coverage/lcov.info packages/*/coverage/lcov.info' './coverage/lcov.info'
+
+
+      # - name: Upload to Codecov
+      #   uses: codecov/codecov-action@v4
+      #   with:
+      #     token: ${{ secrets.CODECOV_TOKEN }}
+      #     files: ./coverage/lcov.info
+      #     fail_ci_if_error: true
```

Comment:
let's just remove for now. If we wan to upload coverage reports we can set that up later.